modules/frontend/metrics_query_handler.go

```diff
@@ -43,6 +43,11 @@ func newQueryInstantStreamingGRPCHandler(cfg Config, next pipeline.AsyncRoundTri
 			End:   req.End,
 			Step:  req.End - req.Start,
 		}
+		// if a limit is being enforced, honor the request if it is less than the limit
+		// else set it to max limit
+		if cfg.Metrics.Sharder.MaxResponseSeries > 0 && (qr.MaxSeries > uint32(cfg.Metrics.Sharder.MaxResponseSeries) || qr.MaxSeries == 0) {
```

Comment:
Do the handlers need to do this too, when the sharder is already doing it?  It looks like doing in the 1 place in the sharder should work.

Reply:
Ah ok.  Let's still consolidate it to 1 function?

Reply:
i think it only needs to be done here? the sharder can assume that the value is set correctly?

although looking at search it looks like we repeat the logic 2x. once in the sharder and once in the handler:

https://github.com/grafana/tempo/blob/main/modules/frontend/search_sharder.go#L82
https://github.com/grafana/tempo/blob/main/modules/frontend/search_handlers.go#L128

it seems like we should establish a better pattern and only do this once, but agree with marty that consolidating the logic in one func is the right choice